Parallel p4c builds recently broken?

I have used a script for building p4c for over a year now, consistently using the ability of make to do parallel builds using command line options like `-j 4` to enable up to 4 parallel processes to be run by `make`.

Since some time around the latest version of p4c as of 2018-Dec-21, or perhaps starting 1 or 2 commits after that date, I get fairly consistent failures when using `-j4`, but it seems that I can get consistent build success if I use `-j1`, or no `-j<number>` argument at all.

Since there is an easy workaround, I do not consider this a high priority issue, but in case anyone who knows more about the p4c build process than I do wants this feature to work, I wanted to create an issue that perhaps it is not.

Below are the particular command line options and steps I have been using that cause the failure:
```
cd p4c
/bin/rm -fr build
mkdir build
cd build
cmake .. -DCMAKE_BUILD_TYPE=DEBUG
make -j4
```

I am using Ubuntu Linux 16.04 and/or 18.04 for my builds, and seeing similar parallel build issues with both.

I reproduce here the error:
```
In file included from /home/jafinger/p4c/frontends/parsers/parserDriver.h:25:0,
                 from /home/jafinger/p4c/frontends/p4/parseAnnotations.h:21,
                 from /home/jafinger/p4c/frontends/p4/frontend.h:21,
                 from /home/jafinger/p4c/backends/bmv2/common/sharedActionSelectorCheck.h:23,
                 from /home/jafinger/p4c/backends/bmv2/common/control.h:30,
                 from /home/jafinger/p4c/backends/bmv2/common/control.cpp:17,
                 from /home/jafinger/p4c/build/backends/bmv2/unified_bmv2_backend_common_srcs_1.cpp:4:
/home/jafinger/p4c/frontends/parsers/p4/abstractP4Lexer.hpp:4:10: fatal error: frontends/parsers/p4/p4parser.hpp: No such file or directory
 #include "frontends/parsers/p4/p4parser.hpp"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/backends/bmv2/CMakeLists.txt
+++ b/backends/bmv2/CMakeLists.txt
@@ -84,7 +84,7 @@ add_cpplint_files (${CMAKE_CURRENT_SOURCE_DIR} "${BMV2_BACKEND_COMMON_SRCS};${BM
 
 build_unified(BMV2_BACKEND_COMMON_SRCS)
 add_library(bmv2backend ${BMV2_BACKEND_COMMON_SRCS})
-add_dependencies(bmv2backend genIR)
+add_dependencies(bmv2backend genIR frontend)
 
 build_unified(BMV2_SIMPLE_SWiTCH_SRCS)
 add_executable(p4c-bm2-ss ${BMV2_SIMPLE_SWITCH_SRCS})

--- a/backends/ebpf/CMakeLists.txt
+++ b/backends/ebpf/CMakeLists.txt
@@ -56,7 +56,7 @@ set (P4C_EBPF_DIST_HEADERS p4include/ebpf_model.p4)
 build_unified(P4C_EBPF_SRCS ALL)
 add_executable(p4c-ebpf ${P4C_EBPF_SRCS})
 target_link_libraries (p4c-ebpf ${P4C_LIBRARIES} ${P4C_LIB_DEPS})
-add_dependencies(p4c-ebpf genIR)
+add_dependencies(p4c-ebpf genIR frontend)
 
 install (TARGETS p4c-ebpf
   RUNTIME DESTINATION ${P4C_RUNTIME_OUTPUT_DIRECTORY})

--- a/backends/graphs/CMakeLists.txt
+++ b/backends/graphs/CMakeLists.txt
@@ -33,7 +33,7 @@ add_cpplint_files(${CMAKE_CURRENT_SOURCE_DIR} "${GRAPHS_SRCS};${GRAPHS_HDRS}")
 build_unified(GRAPHS_SRCS ALL)
 add_executable(p4c-graphs ${GRAPHS_SRCS} ${EXTENSION_P4_14_CONV_SOURCES})
 target_link_libraries (p4c-graphs ${P4C_LIBRARIES} ${P4C_LIB_DEPS})
-add_dependencies(p4c-graphs genIR)
+add_dependencies(p4c-graphs genIR frontend)
 
 install (TARGETS p4c-graphs
   RUNTIME DESTINATION ${P4C_RUNTIME_OUTPUT_DIRECTORY})

--- a/backends/p4test/CMakeLists.txt
+++ b/backends/p4test/CMakeLists.txt
@@ -30,7 +30,7 @@ add_cpplint_files (${CMAKE_CURRENT_SOURCE_DIR} "${P4TEST_SRCS};${P4TEST_HDRS}")
 build_unified(P4TEST_SRCS ALL)
 add_executable(p4test ${P4TEST_SRCS} ${EXTENSION_P4_14_CONV_SOURCES})
 target_link_libraries (p4test ${P4C_LIBRARIES} ${P4C_LIB_DEPS})
-add_dependencies(p4test genIR)
+add_dependencies(p4test genIR frontend)
 
 install (TARGETS p4test
   RUNTIME DESTINATION ${P4C_RUNTIME_OUTPUT_DIRECTORY})

